src/org/omegat/util/StringUtil.java

```diff
@@ -511,4 +511,31 @@ public static String format(String str, Object... arguments) {
         str = str.replaceAll("'", "''");
         return MessageFormat.format(str, arguments);
     }
+
+    /**
+     * convert Zenkaku/Hankaku alphabet to regex  that match both
+     *
+     * replace (\uFF01 - \uFF5E) to (\u0021- \u007E)
+     * such as text = text.replaceAll("u\0021", "(\uFF01|\u0021)");
+     * except for chars that consist of regex.
+     * FIXME: handling 0x20-0x2f, 0x5c
+     *
+     * @param text
+     * @return String with regex
+     */
+    public static String replaceZenHan(String text) {
```

Comment:
I'd like to try a following algorithm;

1. Build half-width version of search text if necessary
2. Build full-width version of search text if possible
3. If (1) and (2) is same, there is no characters to be aware for full/half width, just return original text
4. Escape special characters for regex with `'\'` ex. `'\|'`
5. Build regex `'(<half-width text>|<full-width text>)'` and return.

To implement 1 and 2, I can use a way Okapi's detection of characters incl. Hangul, katakana, etc. 

A method for 4 would be reusable so it can be independent util method, `String escapeRegexChars(String text)`